File: nomor 2/golang/ex11.go
package main

import "fmt"

func generateData() []string {
	const number = "0821234567"
	var customers []string
	var mobileNumber string
	for i := 0; i < 100; i++ {
		if i < 10 {
			mobileNumber = number
		} else {
			mobileNumber = number + fmt.Sprintf("%d", i)
		}
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(array []string) {
	for i := 0; i < len(array); i++ {
		fmt.Println("Sending Promo to", array[i])
	}
	for i := 0; i < 10; i++ {
		fmt.Println("Sending Discount to Chosen Customer", i+1)
	}
}

func main() {
	customers := generateData()
	sendPromoDiscount(customers)
}

File: nomor 2/kotlin/ex11.kt
import kotlin.math.log10
import kotlin.math.pow

fun generateData(): List<String> {
    val number = "0821234567"
    val customers = mutableListOf<String>()
    for (i in 0..99) {
        val mobileNumber = if (i < 10) number else "$number${i}"
        customers.add(mobileNumber)
    }
    return customers
}

fun sendPromoDiscount(array: List<String>) {
    for (mobileNumber in array) {
        println("Sending Promo to $mobileNumber")
    }
    for (i in 1..10) {
        val chosenCustomer = log10(i.toDouble() + 1).pow(3).toInt() % array.size
        println("Sending Discount to Chosen Customer ${chosenCustomer + 1}")
    }
}

fun main() {
    val customers = generateData()
    sendPromoDiscount(customers)
}
